libs/langchain/langchain/schema/runnable/locals.py

```diff
@@ -0,0 +1,156 @@
+        replace: bool = False,
+    ) -> None:
+        if config is None:
+            raise ValueError(
+                "PutLocalVar should only be used in a RunnableSequence, and should "
+                "therefore always receive a non-null config."
+            )
+        if isinstance(self.key, str):
+            if self.key not in config["_locals"] or replace:
+                config["_locals"][self.key] = input
+            else:
+                config["_locals"][self.key] += input
+        elif isinstance(self.key, Mapping):
+            if not isinstance(input, Mapping):
+                raise TypeError(
+                    f"Received key of type Mapping but input of type {type(input)}. "
+                    f"input is expected to be of type Mapping when key is Mapping."
+                )
+            for input_key, put_key in self.key.items():
+                if put_key not in config["_locals"] or replace:
+                    config["_locals"][put_key] = input[input_key]
+                else:
+                    config["_locals"][put_key] += input[input_key]
+        else:
+            raise TypeError(
+                f"`key` should be a string or Mapping[str, str], received type "
+                f"{(type(self.key))}."
+            )
+
+    def invoke(self, input: Input, config: Optional[RunnableConfig] = None) -> Input:
+        self._concat_put(input, config=config, replace=True)
+        return super().invoke(input, config=config)
+
+    async def ainvoke(
+        self,
+        input: Input,
+        config: Optional[RunnableConfig] = None,
+        **kwargs: Optional[Any],
+    ) -> Input:
+        self._concat_put(input, config=config, replace=True)
+        return await super().ainvoke(input, config=config)
+
+    def transform(
+        self,
+        input: Iterator[Input],
+        config: Optional[RunnableConfig] = None,
+        **kwargs: Optional[Any],
+    ) -> Iterator[Input]:
+        for chunk in super().transform(input, config=config):
+            self._concat_put(chunk, config=config)
+            yield chunk
+
+    async def atransform(
+        self,
+        input: AsyncIterator[Input],
+        config: Optional[RunnableConfig] = None,
+        **kwargs: Optional[Any],
+    ) -> AsyncIterator[Input]:
+        async for chunk in super().atransform(input, config=config):
+            self._concat_put(chunk, config=config)
+            yield chunk
+
+
+class GetLocalVar(
+    Serializable, Runnable[Input, Union[Output, Dict[str, Union[Input, Output]]]]
+):
+    key: str
+    """The key to extract from the local state."""
+    passthrough_key: Optional[str] = None
+    """The key to use for passing through the invocation input. 
+    
+    If None, then only the value retrieved from local state is returned. Otherwise a 
+        dictionary ``{self.key: <<retrieved_value>>, self.passthrough_key: <<input>>}``
+        is returned.
+    """
+
+    def __init__(self, key: str, **kwargs: Any) -> None:
```

Comment:
Should this be deleted?

Reply:
This is to allow instantiating with positional arg, which given it's a single arg makes sense